lib/content_publisher/document_collection_group_membership_migrator.rb

```diff
@@ -0,0 +1,28 @@
+module ContentPublisher
+  class DocumentCollectionGroupMembershipMigrator
+    attr_reader :document
+
+    def initialize(document)
+      @document = document
+    end
+
+    def call
+      edition = document.published_edition.present? ? document.published_edition : document.latest_edition
```

Comment:
minor but this could be simplified to `edition = document.published_edition || document.latest_edition`